src/components/routes/Articles/Articles.jsx

```diff
@@ -10,22 +10,17 @@ function Articles() {
 
     useEffect(() => {
         setLoadProgress(15);
-        const queryString = endpoints.articles;
-        fetch(queryString)
-            .then((response) => {
-                setLoadProgress(50);
-                return response.json();
-            })
-            .then((body) => {
-                setArticles(body.articles);
-                setLoadProgress(100);
-                setTimeout(() => {
-                    setLoadProgress(0);
+        getAllArticles()
+            .then((data) => {
+                if (data) {
+                    setArticles(data.articles);
+                    setLoadProgress(100);
                     setTimeout(() => {
+                        setLoadProgress(0);
                         setShowArticles(true);
-                    }, 500); 
-                }, 1250); 
-            });
+                    }, 1000);
```

Comment:
Instead of having a timeout for your loader you could have something that just display as something is loading. this will add some artificial time when loading in the data.